Build faills on 32-bit Fedora Rawhide

**Describe the bug**

```
make[2]: Entering directory '/builddir/build/BUILD/hdf5-1.14.5-build/hdf5-hdf5_1.14.5/build/fortran/test'
+ gfortran -I. -I../../../fortran/test -I../../src -I../../fortran/src -std=f2008 -Waliasing -Wall -Wcharacter-truncation -Wextra -Wimplicit-interface -Wsurprising -Wunderflow -pedantic -Wintrinsics-std -Wimplicit-procedure -Wreal-q-constant -Wfunction-elimination -Wrealloc-lhs -Wrealloc-lhs-all -Wno-c-binding-type -Winteger-division -Wfrontend-loop-interchange -fdiagnostics-urls=never -fno-diagnostics-color -s -Wno-unused-dummy-argument -Wno-array-temporaries -O3 -I../../fortran/src -I../../fortran/src -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m32 -march=i686 -mtune=generic -msse2 -mfpmath=sse -mstackrealign -fasynchronous-unwind-tables -fstack-clash-protection -I/usr/lib/gfortran/modules -c -o tH5T_F03.o ../../../fortran/test/tH5T_F03.F90
../../../fortran/test/tH5T_F03.F90:3009:17:
 3009 |         IF(.NOT.(orig_data(i,j).EQ.orig_data(i,j))) CYCLE  ! skip IF value is NaN
      |                 1
Warning: Equality comparison for REAL(4) at (1) [-Wcompare-reals]
../../../fortran/test/tH5T_F03.F90:1184:81:
 1184 |        CALL verify("h5kind_to_type",dset_data_r31(i),data_out_r31(i),total_error)
      |                                                                                 1
Error: There is no specific subroutine for the generic ‘verify’ at (1)
make[2]: *** [Makefile:1560: tH5T_F03.o] Error 1
make[2]: Leaving directory '/builddir/build/BUILD/hdf5-1.14.5-build/hdf5-hdf5_1.14.5/build/fortran/test'
make[1]: *** [Makefile:919: all-recursive] Error 1
```

**Expected behavior**
Compiles without error

**Platform (please complete the following information)**
 - HDF5 version - 1.14.5
 - OS and version - Fedora Rawhide
 - Compiler and version - gcc/gfortran 14.2.1
 - Build system - autotools
 - Any configure options you specified
 ```
+ CFLAGS='-O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m32 -march=i686 -mtune=generic -msse2 -mfpmath=sse -mstackrealign -fasynchronous-unwind-tables -fstack-clash-protection '
+ export CFLAGS
+ CXXFLAGS='-O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m32 -march=i686 -mtune=generic -msse2 -mfpmath=sse -mstackrealign -fasynchronous-unwind-tables -fstack-clash-protection '
+ export CXXFLAGS
+ FFLAGS='-O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m32 -march=i686 -mtune=generic -msse2 -mfpmath=sse -mstackrealign -fasynchronous-unwind-tables -fstack-clash-protection -I/usr/lib/gfortran/modules '
+ export FFLAGS
+ FCFLAGS='-O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m32 -march=i686 -mtune=generic -msse2 -mfpmath=sse -mstackrealign -fasynchronous-unwind-tables -fstack-clash-protection -I/usr/lib/gfortran/modules '
```
```
+ ../configure --build=i686-redhat-linux-gnu --host=i686-redhat-linux-gnu --program-prefix= --disable-dependency-tracking --prefix=/usr --exec-prefix=/usr --bindir=/usr/bin --sbindir=/usr/sbin --sysconfdir=/etc --datadir=/usr/share --includedir=/usr/include --libdir=/usr/lib --libexecdir=/usr/libexec --localstatedir=/var --runstatedir=/run --sharedstatedir=/var/lib --mandir=/usr/share/man --infodir=/usr/share/info --disable-silent-rules --enable-fortran --enable-hl --enable-shared --with-szlib --enable-cxx --with-default-plugindir=/usr/lib/hdf5/plugin
```
 
**Additional context**
I'm trying to update the Fedora hdf5 package to the 1.14.X branch.  Do you still support 32-bit platforms?  If not, it may just be time for us to drop supporting it as well.

I'm not sure why there seems to be duplicates of the Fortran integer and real checks.  We also seem to end up with duplicates in conftest.c at some point.  From config.log:
```
conftest.c:58:9: warning: "H5CONFIG_F_NUM_RKIND" redefined
   58 | #define H5CONFIG_F_NUM_RKIND INTEGER, PARAMETER :: num_rkinds = 3
      |         ^~~~~~~~~~~~~~~~~~~~
conftest.c:44:9: note: this is the location of the previous definition
   44 | #define H5CONFIG_F_NUM_RKIND INTEGER, PARAMETER :: num_rkinds = 4
      |         ^~~~~~~~~~~~~~~~~~~~
conftest.c:59:9: warning: "H5CONFIG_F_RKIND" redefined
   59 | #define H5CONFIG_F_RKIND INTEGER, DIMENSION(1:num_rkinds) :: rkind = (/4,8,10/)
      |         ^~~~~~~~~~~~~~~~
conftest.c:46:9: note: this is the location of the previous definition
   46 | #define H5CONFIG_F_RKIND INTEGER, DIMENSION(1:num_rkinds) :: rkind = (/4,8,10,16/)
      |         ^~~~~~~~~~~~~~~~
```
[config.log](https://github.com/user-attachments/files/17251594/config.log)

--- a/configure.ac
+++ b/configure.ac
@@ -854,6 +854,7 @@ if test "X$HDF_FORTRAN" = "Xyes"; then
   if test "$MY_FLT128_DIG" -gt "$MY_LDBL_DIG" ; then
     AC_MSG_RESULT([yes])
     AC_DEFINE([HAVE_FLOAT128], [1], [Determine if __float128 will be used in the Fortran wrappers])
+    AC_CHECK_SIZEOF([__float128])
   else
     ## Can't use __float128, but write an undef line anyway
     AC_MSG_RESULT([no])

